esp/services/ws_workunits/ws_wuresult.cpp

```diff
@@ -0,0 +1,441 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2020 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "ws_workunits_esp.ipp"
+#include "exception_util.hpp"
+#include "ws_wuresult.hpp"
+#include "ws_workunitsHelpers.hpp"
+
+FlushingWUResultBuffer::~FlushingWUResultBuffer()
+{
+    try
+    {
+        if (!outIOS)
+        {
+            if (!s.isEmpty())
+                response->sendChunk(s);
+        }
+    }
+    catch (IException *e)
+    {
+        // Ignore any socket errors that we get at termination - nothing we can do about them anyway...
+        e->Release();
+    }
+}
+
+void FlushingWUResultBuffer::flushXML(StringBuffer& current, bool closing)
+{
+    if (outIOS)
+    {
+        outIOS->write(current.length(), current.str());
+        current.clear();
+        return;
+    }
+
+    s.append(current);
+    current.clear();
+    if (s.length() < (closing ? 1 : wuResultDownloadFlushThreshold))
```

Comment:
Minor, but might be a slightly clearer if this was broken into two lines.  
unsigned threshold = closing ? 1 : wuResultDownloadFlushThreshold;